// @bilibili/userAvatar/dist/user-avatar.js
/*!
 * @bilibili/userAvatar
 * version: 1.0.3. Powered by main-frontend
 * 用户头像公共组件.
 * author: wuxiuran
 */
module.exports = function(t) {
    var e = {};

    function n(r) {
        if (e[r]) return e[r].exports;
        var o = e[r] = {
            i: r,
            l: !1,
            exports: {}
        };
        return t[r].call(o.exports, o, o.exports, n), o.l = !0, o.exports
    }
    return n.m = t, n.c = e, n.d = function(t, e, r) {
        n.o(t, e) || Object.defineProperty(t, e, {
            enumerable: !0,
            get: r
        })
    }, n.r = function(t) {
        "undefined" != typeof Symbol && Symbol.toStringTag && Object.defineProperty(t, Symbol.toStringTag, {
            value: "Module"
        }), Object.defineProperty(t, "__esModule", {
            value: !0
        })
    }, n.t = function(t, e) {
        if (1 & e && (t = n(t)), 8 & e) return t;
        if (4 & e && "object" == typeof t && t && t.__esModule) return t;
        var r = Object.create(null);
        if (n.r(r), Object.defineProperty(r, "default", {
                enumerable: !0,
                value: t
            }), 2 & e && "string" != typeof t)
            for (var o in t) n.d(r, o, function(e) {
                return t[e]
            }.bind(null, o));
        return r
    }, n.n = function(t) {
        var e = t && t.__esModule ? function() {
            return t.default
        } : function() {
            return t
        };
        return n.d(e, "a", e), e
    }, n.o = function(t, e) {
        return Object.prototype.hasOwnProperty.call(t, e)
    }, n.p = "", n(n.s = 16)
}([function(t, e, n) {
    "use strict";
    (function(t) {
        function r(t) {
            return (r = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(t) {
                return typeof t
            } : function(t) {
                return t && "function" == typeof Symbol && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t
            })(t)
        }
        n.d(e, "f", (function() {
            return o
        })), n.d(e, "b", (function() {
            return a
        })), n.d(e, "d", (function() {
            return s
        })), n.d(e, "a", (function() {
            return f
        })), n.d(e, "e", (function() {
            return b
        })), n.d(e, "c", (function() {
            return d
        }));
        var o = "object" == ("undefined" == typeof self ? "undefined" : r(self)) && self.self === self && self || "object" == (void 0 === t ? "undefined" : r(t)) && t.global === t && t || void 0,
            i = o.devicePixelRatio && 3 === o.devicePixelRatio ? 3 : 2,
            a = function(t) {
                o.document && (o.document.addEventListener ? o.document.addEventListener("DOMContentLoaded", (function() {
                    t && t()
                })) : o.document.onreadystatechange = function() {
                    "complete" === o.document.readyState && t && t()
                })
            },
            c = function(t) {
                if (!t) return "";
                var e = t.replace(/^(http:\/\/|:\/\/|\/\/|\/)/, "https://");
                return e.match(/^https:\/\//) ? e : "https://" + e
            },
            s = function(t, e, n) {
                var r = arguments.length > 3 && void 0 !== arguments[3] && arguments[3];
                return function(t) {
                    if (!t) return t;
                    var o = (t = c(t)).match(/(.*\.(jpg|jpeg|gif|png|JPG|JPEG|GIF|PNG|webp|WEBP))(\?.*)?/),
                        a = -1 != t.indexOf("/bfs/");
                    try {
                        if (!o || "gif" === o[2] || !a || !o) return t
                    } catch (e) {
                        return t
                    }
                    var s = e && n ? "@" + Math.round(+e * i) + "w_" + Math.round(+n * i) + "h" : "@";
                    r && (s += "_1c");
                    var l = o[3] ? o[3] : "";
                    return u ? o[1] + s + ".webp" + l : o[1] + s + "." + o[2] + l
                }(t)
            },
            u = function() {
                try {
                    return o.document && 0 == o.document.createElement("canvas").toDataURL("image/webp").indexOf("data:image/webp")
                } catch (t) {
                    return !1
                }
            }(),
            l = function(t, e) {
                return new RegExp("\\b" + e + "\\b").test(t.className)
            },
            f = function(t, e) {
                t.className ? l(t, e) || (t.className += " " + e) : t.className = e
            },
            b = function(t, e) {
                var n = new RegExp("\\b" + e + "\\b", "g");
                l(t, e) && (t.className = t.className.replace(n, "").replace(/^\s+|\s+$/g, "").replace(/\s+/g, " "))
            },
            d = function(t) {
                return t = t <= 48 ? 48 : t <= 60 ? 60 : t <= 80 ? 80 : 120
            }
    }).call(this, n(4))
}, function(t, e) {
    function n(t) {
        return (n = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(t) {
            return typeof t
        } : function(t) {
            return t && "function" == typeof Symbol && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t
        })(t)
    }
    t.exports = function(t) {
        return null != t && "object" == n(t)
    }
}, function(t, e, n) {
    var r = n(1),
        o = n(8);
    t.exports = function(t) {
        return r(t) && 1 === t.nodeType && !o(t)
    }
}, function(t, e, n) {
    var r = n(5),
        o = n(15),
        i = n(1);
    t.exports = function(t) {
        return "string" == typeof t || !o(t) && i(t) && "[object String]" == r(t)
    }
}, function(t, e) {
    function n(t) {
        return (n = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(t) {
            return typeof t
        } : function(t) {
            return t && "function" == typeof Symbol && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t
        })(t)
    }
    var r;
    r = function() {
        return this
    }();
    try {
        r = r || new Function("return this")()
    } catch (t) {
        "object" === ("undefined" == typeof window ? "undefined" : n(window)) && (r = window)
    }
    t.exports = r
}, function(t, e, n) {
    var r = n(6),
        o = n(11),
        i = n(12),
        a = r ? r.toStringTag : void 0;
    t.exports = function(t) {
        return null == t ? void 0 === t ? "[object Undefined]" : "[object Null]" : a && a in Object(t) ? o(t) : i(t)
    }
}, function(t, e, n) {
    var r = n(9).Symbol;
    t.exports = r
}, function(t, e, n) {}, function(t, e, n) {
    var r = n(5),
        o = n(13),
        i = n(1),
        a = Function.prototype,
        c = Object.prototype,
        s = a.toString,
        u = c.hasOwnProperty,
        l = s.call(Object);
    t.exports = function(t) {
        if (!i(t) || "[object Object]" != r(t)) return !1;
        var e = o(t);
        if (null === e) return !0;
        var n = u.call(e, "constructor") && e.constructor;
        return "function" == typeof n && n instanceof n && s.call(n) == l
    }
}, function(t, e, n) {
    function r(t) {
        return (r = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(t) {
            return typeof t
        } : function(t) {
            return t && "function" == typeof Symbol && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t
        })(t)
    }
    var o = n(10),
        i = "object" == ("undefined" == typeof self ? "undefined" : r(self)) && self && self.Object === Object && self,
        a = o || i || Function("return this")();
    t.exports = a
}, function(t, e, n) {
    (function(e) {
        function n(t) {
            return (n = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(t) {
                return typeof t
            } : function(t) {
                return t && "function" == typeof Symbol && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t
            })(t)
        }
        var r = "object" == (void 0 === e ? "undefined" : n(e)) && e && e.Object === Object && e;
        t.exports = r
    }).call(this, n(4))
}, function(t, e, n) {
    var r = n(6),
        o = Object.prototype,
        i = o.hasOwnProperty,
        a = o.toString,
        c = r ? r.toStringTag : void 0;
    t.exports = function(t) {
        var e = i.call(t, c),
            n = t[c];
        try {
            t[c] = void 0;
            var r = !0
        } catch (t) {}
        var o = a.call(t);
        return r && (e ? t[c] = n : delete t[c]), o
    }
}, function(t, e) {
    var n = Object.prototype.toString;
    t.exports = function(t) {
        return n.call(t)
    }
}, function(t, e, n) {
    var r = n(14)(Object.getPrototypeOf, Object);
    t.exports = r
}, function(t, e) {
    t.exports = function(t, e) {
        return function(n) {
            return t(e(n))
        }
    }
}, function(t, e) {
    var n = Array.isArray;
    t.exports = n
}, function(t, e, n) {
    "use strict";
    n.r(e);
    var r, o = n(0),
        i = function(t, e) {
            var n = t.src,
                r = t.height,
                i = t.width,
                a = t.className,
                c = t.style,
                s = t.radius;
            return "<img ".concat(c ? ' style="' + c + '"' : "", ' class="').concat(a, '" ').concat(e ? "data-" : "", 'src="').concat(Object(o.d)(n, i, r, s), '" alt />')
        },
        a = (n(7), n(2)),
        c = n.n(a),
        s = n(3),
        u = n.n(s),
        l = ((r = o.f).LazyLoad = function(t) {
            var e = {
                src: "data-src",
                srcset: "data-srcset",
                selector: ".lazyload",
                root: null,
                rootMargin: "0px",
                threshold: 0
            };

            function n(t, n) {
                this.settings = function t() {
                    var e = {},
                        n = !1,
                        r = 0,
                        o = arguments.length;
                    "[object Boolean]" === Object.prototype.toString.call(arguments[0]) && (n = arguments[0], r++);
                    for (var i = function(r) {
                            for (var o in r) Object.prototype.hasOwnProperty.call(r, o) && (n && "[object Object]" === Object.prototype.toString.call(r[o]) ? e[o] = t(!0, e[o], r[o]) : e[o] = r[o])
                        }; r < o; r++) {
                        var a = arguments[r];
                        i(a)
                    }
                    return e
                }(e, n || {}), this.images = t || document.querySelectorAll(this.settings.selector), this.observer = null, this.init()
            }
            if (n.prototype = {
                    init: function() {
                        if (t.IntersectionObserver) {
                            var e = this,
                                n = {
                                    root: this.settings.root,
                                    rootMargin: this.settings.rootMargin,
                                    threshold: [this.settings.threshold]
                                };
                            this.observer = new IntersectionObserver((function(t) {
                                Array.prototype.forEach.call(t, (function(t) {
                                    if (t.intersectionRatio > 0) {
                                        e.observer.unobserve(t.target);
                                        var n = t.target.getAttribute(e.settings.src),
                                            r = t.target.getAttribute(e.settings.srcset);
                                        "img" === t.target.tagName.toLowerCase() ? (n && (t.target.src = n), r && (t.target.srcset = r)) : t.target.style.backgroundImage = "url(" + n + ")"
                                    }
                                }))
                            }), n), Array.prototype.forEach.call(this.images, (function(t) {
                                e.observer.observe(t)
                            }))
                        } else this.loadImages()
                    },
                    loadAndDestroy: function() {
                        this.settings && (this.loadImages(), this.destroy())
                    },
                    loadImages: function() {
                        if (this.settings) {
                            var t = this;
                            Array.prototype.forEach.call(this.images, (function(e) {
                                var n = e.getAttribute(t.settings.src),
                                    r = e.getAttribute(t.settings.srcset);
                                "img" === e.tagName.toLowerCase() ? (n && (e.src = n), r && (e.srcset = r)) : e.style.backgroundImage = "url('" + n + "')"
                            }))
                        }
                    },
                    destroy: function() {
                        this.settings && (this.observer.disconnect(), this.settings = null)
                    }
                }, t.lazyload = function(t, e) {
                    return new n(t, e)
                }, t.jQuery) {
                var r = t.jQuery;
                r.fn.lazyload = function(t) {
                    return (t = t || {}).attribute = t.attribute || "data-src", new n(r.makeArray(this), t), this
                }
            }
            return n
        }(r), r.LazyLoad);

    function f(t, e) {
        for (var n = 0; n < e.length; n++) {
            var r = e[n];
            r.enumerable = r.enumerable || !1, r.configurable = !0, "value" in r && (r.writable = !0), Object.defineProperty(t, r.key, r)
        }
    }
    var b = function(t) {
        return t.querySelectorAll(".bili-avatar-img").forEach((function(t) {
            t.addEventListener("error", (function(t) {
                Object(o.a)(t.target, "bili-avatar-img-error")
            }))
        }))
    };
    Object(o.b)((function() {
        return b(document)
    }));
    var d = function() {
        function t() {
            ! function(t, e) {
                if (!(t instanceof e)) throw new TypeError("Cannot call a class as a function")
            }(this, t), this.lazy = [], this.$refs = {}
        }
        var e, n, r;
        return e = t, (n = [{
            key: "actionFn",
            value: function(t) {
                this.lazy.length && this.lazy.forEach((function(e) {
                    e[t]()
                }))
            }
        }, {
            key: "loadImages",
            value: function() {
                this.actionFn("loadImages")
            }
        }, {
            key: "loadAndDestroy",
            value: function() {
                this.actionFn("loadAndDestroy")
            }
        }, {
            key: "append",
            value: function(t, e, n) {
                var r = t.avatar,
                    a = t.pendent,
                    s = t.avatarWidth,
                    f = t.iconType,
                    d = t.isAnimate,
                    p = t.duration,
                    y = t.customIcon,
                    v = function(t, e) {
                        var n = t.avatar,
                            r = void 0 === n ? "" : n,
                            a = t.pendent,
                            c = void 0 === a ? "" : a,
                            s = t.avatarHeight,
                            u = t.avatarWidth,
                            l = t.pendentHeight,
                            f = t.pendentWidth,
                            b = t.iconType,
                            d = void 0 === b ? "" : b,
                            p = t.isAnimate,
                            y = void 0 !== p && p,
                            v = t.duration,
                            m = void 0 === v ? 2 : v,
                            g = t.customIcon,
                            h = void 0 === g ? "" : g;
                        return '\n    <div class="bili-avatar">\n      '.concat(i({
                            src: r,
                            className: "bili-avatar-img bili-avatar-face bili-avatar-img-radius",
                            height: s,
                            width: u,
                            radius: !0
                        }, e), "\n      \n      ").concat(c ? '<div class="bili-avatar-pendent">\n          '.concat(i({
                            src: c,
                            className: "bili-avatar-img",
                            style: y && "-webkit-animation: bili-avatar ".concat(m, "s steps(").concat(39, ") infinite; -moz-animation: bili-avatar ").concat(m, "s steps(").concat(39, ") infinite; animation: bili-avatar ").concat(m, "s steps(").concat(39, ") infinite;"),
                            height: l,
                            width: f
                        }, e), "\n        </div>") : "", '\n      <span\n        class="bili-avatar-icon').concat(d && !h ? " bili-avatar-icon--" + d : "", '"\n        ').concat(h ? 'style="background-image: url(' + Object(o.d)(h, u / 3, u / 3, !0) + ');"' : "", "\n      ></span>\n    </div>\n  ")
                    }({
                        avatar: r,
                        pendent: a,
                        avatarHeight: s = Object(o.c)(s),
                        avatarWidth: s,
                        pendentHeight: 1.7 * s,
                        pendentWidth: d ? 1.7 * s * 40 : 1.7 * s,
                        iconType: f,
                        isAnimate: d,
                        duration: p,
                        customIcon: y
                    }, e),
                    m = e;
                if (!m) return v;
                c()(m) ? (m.innerHTML = v, b(m)) : u()(m) && ((m = document.querySelector(m)).innerHTML = v, b(m)), this.lazy.push(new l(null, {
                    root: null,
                    rootMargin: "0px",
                    threshold: 0,
                    selector: ".bili-avatar-img"
                })), n && (this.$refs[n] = m.firstElementChild)
            }
        }, {
            key: "resetConfig",
            value: function(t, e) {
                var n = e.avatar,
                    r = e.pendent,
                    a = e.avatarWidth,
                    s = e.iconType,
                    l = e.isAnimate,
                    f = e.duration,
                    b = e.customIcon;
                a = Object(o.c)(a);
                var d = null;
                if (c()(t) ? d = t : u()(t) && (d = this.$refs[t]), d) {
                    var p = d.querySelector(".bili-avatar-pendent img");
                    if (p) r ? (p.src = Object(o.d)(r, l ? 1.7 * a * 40 : 1.7 * a, 1.7 * a), p.addEventListener("load", (function t() {
                        var e = "bili-avatar ".concat(f, "s steps(").concat(39, ") infinite");
                        p.style.cssText = l && "-webkit-animation: ".concat(e, "; -moz-animation: ").concat(e, "; animation: ").concat(e, ";"), p.removeEventListener("load", t)
                    }))) : d.removeChild(d.querySelector(".bili-avatar-pendent"));
                    else if (r) {
                        var y = document.createElement("div");
                        Object(o.a)(y, "bili-avatar-pendent");
                        var v = "bili-avatar ".concat(f, "s steps(").concat(39, ") infinite");
                        y.innerHTML = i({
                            src: r,
                            className: "bili-avatar-img",
                            style: l && "-webkit-animation: ".concat(v, "; -moz-animation: ").concat(v, "; animation: ").concat(v, ";"),
                            height: 1.7 * a,
                            width: l ? 1.7 * a * 40 : 1.7 * a
                        }, !1), d.insertBefore(y, d.querySelector(".bili-avatar-icon"))
                    }
                    var m = d.querySelector(".bili-avatar-icon");
                    b ? (Object(o.e)(m, "bili-avatar-icon--big-vip"), Object(o.e)(m, "bili-avatar-icon--personal"), Object(o.e)(m, "bili-avatar-icon--business"), Object(o.e)(m, "bili-avatar-icon--small-vip"), m.style.backgroundImage = "url(".concat(Object(o.d)(b, a / 3, a / 3, !0), ")")) : (s ? new RegExp(s).test(m.className) || (Object(o.e)(m, "bili-avatar-icon--big-vip"), Object(o.e)(m, "bili-avatar-icon--personal"), Object(o.e)(m, "bili-avatar-icon--business"), Object(o.e)(m, "bili-avatar-icon--small-vip"), Object(o.a)(m, "bili-avatar-icon--".concat(s))) : (Object(o.e)(m, "bili-avatar-icon--big-vip"), Object(o.e)(m, "bili-avatar-icon--personal"), Object(o.e)(m, "bili-avatar-icon--business"), Object(o.e)(m, "bili-avatar-icon--small-vip")), m.style.cssText = "");
                    var g = d.querySelector(".bili-avatar-img");
                    n && (g.src = Object(o.d)(n, a, a, !0))
                }
            }
        }]) && f(e.prototype, n), r && f(e, r), t
    }();
    e.default = d
}]);
